File: packages/task2/Position.js
export class Position {
  constructor() {
    this.horizontal = 0;
    this.depth = 0;
    this.aim = 0;
  }

  moveAim(movement) {
    if (movement.direction === "down") {
      this.aim += movement.steps;
    } else {
      this.aim -= movement.steps;
    }
  }

  move(movement) {
    console.log(this, movement.steps);
    if (["up", "down"].includes(movement.direction)) {
      this.moveAim(movement);
    } else {
      this.horizontal += movement.steps;
      this.depth += movement.steps * this.aim;
    }
  }

  position() {
    return Math.abs(this.depth) * Math.abs(this.horizontal);
  }
}

File: packages/rust/src/task2/position.rs
pub struct Position {
    x: i32,
    y: i32,
}

impl Position {
    pub fn new() -> Position {
        Position {
            x: 0,
            y: 0,
        }
    }

    pub fn move_horizontal(&mut self, steps: i32) {
        self.x += steps
    }

    pub fn move_vertical(&mut self, steps: i32) {
        self.y += steps
    }

    pub fn get_absolute_position(&self) -> i32 {
        return i32::abs(self.x) * i32::abs(self.y)
    }
}